azurerm/data_source_virtual_hub.go

```diff
@@ -128,34 +128,38 @@ func dataSourceArmVirtualHubRead(d *schema.ResourceData, meta interface{}) error
 	}
 	if props := resp.VirtualHubProperties; props != nil {
 		d.Set("address_prefix", props.AddressPrefix)
-		if props.VirtualWan != nil {
-			if err := d.Set("virtual_wan_id", props.VirtualWan.ID); err != nil {
-				return fmt.Errorf("Error setting `virtual_wan_id`: %+v", err)
-			}
-		}
-		if props.VpnGateway != nil {
-			if err := d.Set("s2s_vpn_gateway_id", props.VpnGateway.ID); err != nil {
-				return fmt.Errorf("Error setting `s2s_vpn_gateway_id`: %+v", err)
-			}
+
+		var expressRouteGatewayId *string
+		if props.ExpressRouteGateway != nil {
+			expressRouteGatewayId = props.ExpressRouteGateway.ID
 		}
+		d.Set("express_route_gateway_id", expressRouteGatewayId)
+
+		var p2sVpnGatewayId *string
 		if props.P2SVpnGateway != nil {
-			if err := d.Set("p2s_vpn_gateway_id", props.P2SVpnGateway.ID); err != nil {
-				return fmt.Errorf("Error setting `p2s_vpn_gateway_id`: %+v", err)
-			}
+			p2sVpnGatewayId = props.P2SVpnGateway.ID
 		}
-		if props.ExpressRouteGateway != nil {
-			if err := d.Set("express_route_gateway_id", props.ExpressRouteGateway.ID); err != nil {
-				return fmt.Errorf("Error setting `express_route_gateway_id`: %+v", err)
-			}
+		d.Set("p2s_vpn_gateway_id", p2sVpnGatewayId)
+
+		if err := d.Set("route", flattenArmVirtualHubRoute(props.RouteTable)); err != nil {
+			return fmt.Errorf("Error setting `route`: %+v", err)
 		}
+
+		var vpnGatewayId *string
+		if props.VpnGateway != nil {
```

Comment:
same as above. Gateway fields should be readonly since gateway fields here is associated with virtual hub after double confirmed with service owner of this api. see my another PR #4964